datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -181,7 +239,7 @@ fn coerced_from<'a>(
         Int64
             if matches!(
                 type_from,
-                Null | Int8 | Int16 | Int32 | Int64 | UInt8 | UInt16 | UInt32
+                Null | Int8 | Int16 | Int32 | Int64 | UInt8 | UInt16 | UInt32 | Boolean
```

Comment:
Only i64 is fix for passing existing test in array.slt. More type should be fixed in https://github.com/apache/arrow-datafusion/issues/8302